src/Microsoft.ML.TimeSeries/Deseasonality/MeanDeseasonality.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+
+namespace Microsoft.ML.TimeSeries.Deseasonality
+{
+    internal sealed class MeanDeseasonality : DeseasonalityBase
+    {
+        private double[] _circularComponent;
+
+        public override void Deseasonality(ref double[] values, int period, ref double[] results)
+        {
+            Array.Resize(ref _circularComponent, period);
+
+            var length = values.Length;
+
+            // initialize the circurlar component to 0.
```

Comment:
circular

Reply:
fixed.

---
In reply to: [442525384](https://github.com/dotnet/machinelearning/pull/5202#discussion_r442525384) [](ancestors = 442525384)